### PR TITLE
Feature/iesp 23

### PR DESCRIPTION
**Pull request by melissa-dellorco**

### CODE REVIEW

File: docs/openapi/probing-service-v1.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  description: Development environment
+tags:
+  - name: InteropProbing
+    description: >-
+      Read data operations.
+
+paths:
+   '/interop/probing/v1':
+    get:
+      summary: Monitoring the status of an e-services
```
Comment (galales) [inline docs/openapi/probing-service-v1.yaml]:
```suggestion
      summary: Provides the status of the E-Service
```

File: pom.xml [hunk 2]
```diff
@@ -83,13 +113,124 @@
...
 		</dependency>
+		<dependency>
+			<groupId>org.openapitools</groupId>
+			<artifactId>jackson-databind-nullable</artifactId>
+			<version>0.2.1</version>
+		</dependency>
+		<dependency>
+			<groupId>io.springfox</groupId>
+			<artifactId>springfox-swagger2</artifactId>
+			<version>2.8.0</version>
```
Comment (nttdata-rtorsoli) [inline pom.xml]: Use properties

File: src/main/java/it/pagopa/interop/probing/caller/annotations/ValidateStringArraySize.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import it.pagopa.interop.probing.caller.annotations.validator.StringArrayValidator;
+
+@Target({ ElementType.FIELD })
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = StringArrayValidator.class)
+public @interface ValidateStringArraySize {
+
+	int maxSize();
+
+	String message() default "One of the strings of the array is more than {maxSize} characters long";
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/caller/annotations/ValidateStringArraySize.java:19]: Prepare for Internazionalization i18n of messages

File: src/main/java/it/pagopa/interop/probing/caller/consumer/PollingReceiver.java [hunk 4]
```diff
@@ -16,15 +21,34 @@ public class PollingReceiver {
   @Autowired
   ObjectMapper mapper;
 
+  @Autowired
+  PollingResultSend pollingResultSend;
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/caller/consumer/PollingReceiver.java]: Use private

File: src/main/java/it/pagopa/interop/probing/caller/dto/TelemetryDto.java [hunk 5]
```diff
@@ -1,27 +1,41 @@
...
-  private Integer responseTime;
+  @NotNull(message = "must not be null")
+  private Long responseTime;
 
   @JsonProperty("koReason")
+  @Size(min = 1, max = 2048, message = "must be at least 1 and less than or equal to 2048")
   private String koReason;
 
   @JsonProperty("checkTime")
+  @NotBlank(message = "must not be blank")
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/caller/dto/TelemetryDto.java:37]: See above for messages i18n

**Review by nttdata-rtorsoli — DISMISSED** [review]

File: src/main/resources/logback.xml [hunk 6]
```diff
@@ -39,12 +39,23 @@
 
 	<logger name="org.springframework.boot" level="WARN" />
 	<logger name="com.amazonaws" level="WARN" />
-	<logger name="org.springframework.cloud" level="WARN" />
 	<logger name="io.awspring.cloud" level="WARN" />
 	<logger name="org.projectlombok" level="WARN" />
+	<logger name="org.junit.jupiter" level="WARN" />
+	<logger name="org.mockito" level="WARN" />
```
Comment (galales) [inline src/main/resources/logback.xml]: are these required in non-test package?

File: src/main/java/it/pagopa/interop/probing/caller/util/constant/ProjectConstants.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package it.pagopa.interop.probing.caller.util.constant;
+
+public class ProjectConstants {
+
+  private ProjectConstants() {
+    super();
+  }
+
+  public static final String SQS_TELEMETRY_RESULT = "Telemetry-result-queue";
+  public static final String SQS_POLLING_QUEUE = "Poll-result-queue";
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/caller/util/constant/ProjectConstants.java]: Queue names should be passed from external configurations
```suggestion
```

File: src/main/java/it/pagopa/interop/probing/caller/annotations/ValidateStringArraySize.java (hunk 3 above)
Comment (gdellorusso) [reply src/main/java/it/pagopa/interop/probing/caller/annotations/ValidateStringArraySize.java:19]: for now, internationalisation has only been planned for the FE side, but we can discuss it for the BE side

File: src/main/java/it/pagopa/interop/probing/caller/dto/TelemetryDto.java (hunk 5 above)
Comment (gdellorusso) [reply src/main/java/it/pagopa/interop/probing/caller/dto/TelemetryDto.java:37]: same as above

File: src/main/java/it/pagopa/interop/probing/caller/consumer/PollingReceiver.java (hunk 4 above)
Comment (melissa-dellorco) [reply src/main/java/it/pagopa/interop/probing/caller/consumer/PollingReceiver.java]: done

File: pom.xml (hunk 2 above)
Comment (melissa-dellorco) [reply pom.xml]: done

File: src/main/java/it/pagopa/interop/probing/caller/consumer/PollingReceiver.java [hunk 8]
```diff
@@ -16,15 +21,35 @@ public class PollingReceiver {
...
 
     EserviceContentDto service = mapper.readValue(message, EserviceContentDto.class);
 
-    logger.logMessageReceiver(service.getEserviceRecordId());
+    try {
+
+      telemetryResultSend.sendMessage(clientUtil.callProbing(service));
+      pollingResultSend
+          .sendMessage(PollingDto.builder().eserviceRecordId(service.eserviceRecordId())
+              .responseReceived(OffsetDateTime.now()).build());
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/caller/consumer/PollingReceiver.java]: `now()` needs the timezone

File: src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+    logger.logMessageResponseCallProbing(telemetryResult);
+
+    return telemetryResult;
+  }
+
+  private TelemetryDto callRest(TelemetryDto telemetryResult, EserviceContentDto service,
+      long before) throws IOException {
+    Response response = restClientConfig.feignRestClient()
+        .probing(URI.create(Objects.nonNull(service.basePath()) ? service.basePath()[0] : null));
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java]: an empty `basePath` (that should not happen here) would be intercepted and registered as eservice `KO`, which is not correct

File: src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+    Problem problem = new ObjectMapper().readValue(response.body().toString(), Problem.class);
+    logger.logResultCallProbing(response.status(), response.body().toString());
+    return receiverResponse(response.status(), telemetryResult, problem.getDetail(), before);
+  }
+
+  private TelemetryDto callSoap(TelemetryDto telemetryResult, EserviceContentDto service,
+      long before) {
+    ObjectFactory o = new ObjectFactory();
+    ProbingResponse response = soapClientConfig.feignSoapClient().probing(
+        URI.create(Objects.nonNull(service.basePath()) ? service.basePath()[0] : null),
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java]: same

File: src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+  private Logger logger;
+
+  public TelemetryDto callProbing(EserviceContentDto service) {
+
+    TelemetryDto telemetryResult =
+        TelemetryDto.builder().eserviceRecordId(service.eserviceRecordId()).build();
+
+    logger.logMessageCallProbing(service.technology().getValue(), service.basePath()[0]);
+
+    long before = System.currentTimeMillis();
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java]: this should be right before the call, and the calculation should be right after the call

File: docs/openapi/probing-service-v1.yaml [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+      description: >-
+        Checks an e-service status
```
Comment (galales) [inline docs/openapi/probing-service-v1.yaml]:
```suggestion
      description: Provides the status of the E-Service
```

**Comment by galales** [comment]
The JWT sent to the EService will be handled in a different PR?

File: src/main/java/it/pagopa/interop/probing/caller/util/constant/ProjectConstants.java (hunk 7 above)
Comment (gdellorusso) [reply src/main/java/it/pagopa/interop/probing/caller/util/constant/ProjectConstants.java]: changed the constants for better readability of the logs

File: src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java (hunk 9 above)
Comment (gdellorusso) [reply src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java]: what can we do? do we delete the control? it was inserted as an additional control despite the fact that it is expected to be populated

File: src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java (hunk 10 above)
Comment (gdellorusso) [reply src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java]: same

**Review by nttdata-rtorsoli — DISMISSED** [review]

File: src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java (hunk 9 above)
Comment (galales) [reply src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java]: fine by me if you keep the check, but it should be recognized as an internal error and not an EService malfunction

Comment (gdellorusso) [reply src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java]: so easier to eliminate it, done.

**Comment by gdellorusso** [comment]
Added the result of the response call to the polling-result-queue as agreed in this morning's call for display in FE masks

**Review by nttdata-rtorsoli — DISMISSED** [review]

File: src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java [hunk 13]
```diff
@@ -58,28 +57,28 @@ public TelemetryDto callProbing(EserviceContentDto service) {
 
   private TelemetryDto callRest(TelemetryDto telemetryResult, EserviceContentDto service,
       long before) throws IOException {
-    Response response = restClientConfig.feignRestClient()
-        .probing(URI.create(Objects.nonNull(service.basePath()) ? service.basePath()[0] : null));
+    Response response =
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/caller/util/ClientUtil.java:59]: why not moving the `long before = System.currentTimeMillis();` here?

File: src/main/java/it/pagopa/interop/probing/caller/producer/PollingResultSend.java [hunk 14]
```diff
@@ -32,7 +32,6 @@ public void sendMessage(PollingDto polling) throws IOException {
     SendMessageRequest sendMessageRequest = new SendMessageRequest().withQueueUrl(sqsUrl)
         .withMessageBody(objectMapper.writeValueAsString(polling));
     amazonSQS.sendMessage(sendMessageRequest);
-    logger.logMessagePushedToQueue(polling.eserviceRecordId(), ProjectConstants.SQS_POLLING_QUEUE,
-        ProjectConstants.SERVICE_GROUP_ID);
+    logger.logMessagePushedToQueue(polling.eserviceRecordId(), ProjectConstants.SQS_POLLING_QUEUE);
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/caller/producer/PollingResultSend.java]: `ProjectConstants.SQS_POLLING_QUEUE` is not the queue name

File: src/main/java/it/pagopa/interop/probing/caller/producer/TelemetryResultSend.java [hunk 15]
```diff
@@ -33,6 +33,6 @@ public void sendMessage(TelemetryDto service) throws IOException {
         .withMessageBody(objectMapper.writeValueAsString(service));
     amazonSQS.sendMessage(sendMessageRequest);
     logger.logMessagePushedToQueue(service.eserviceRecordId(),
-        ProjectConstants.SQS_TELEMETRY_RESULT, ProjectConstants.SERVICE_GROUP_ID);
+        ProjectConstants.SQS_TELEMETRY_RESULT);
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/caller/producer/TelemetryResultSend.java]: `ProjectConstants. SQS_TELEMETRY_RESULT` is not the queue name

File: src/main/resources/application.properties
Comment (galales) [inline src/main/resources/application.properties:1]: test values committed

Comment (gdellorusso) [reply src/main/resources/application.properties:1]: fixed

**Review by galales — APPROVED** [review]

**Review by nttdata-rtorsoli — APPROVED** [review]

**Pull request merged**